src/unnest.c

```diff
@@ -0,0 +1,226 @@
+#include "data.table.h"
+#include <Rdefines.h>
+
+SEXP unnest(SEXP x) {
+  int n = LENGTH(VECTOR_ELT(x, 0));
+  int p = LENGTH(x);
+  int row_counts[n];
+  SEXPTYPE col_types[p];
+
+  bool all_atomic = true;
+
+  for (int j=0; j<p; j++) {
+    SEXPTYPE this_col = TYPEOF(VECTOR_ELT(x, j));
+    switch(this_col) {
+    case VECSXP :
+      // don't break; drop through to record col_types
```

Comment:
Not writing col_types will keep garbage for this i. I would do loop to write all types, no need switch.

Reply:
I think writing will still happen since there's no `break`, `VECSXP` will continue "falling through" down to L24. Only in `default` there's no writing but we stop then anyway.

Anyway, I'm thinking to switch to `(x, cols)` signature which would obviate this loop.